EIPS/eip-5094.md

```diff
@@ -0,0 +1,105 @@
+
+`rpc_url` is represented as an array of RPC URLs. A minimum of 1 `rpc_url` MUST be present, in the format of `rpc_url=https%3A%2F%2Fpolygon-rpc.com`, or when multiple present `rpc_url=https%3A%2F%2Fpolygon-rpc.com&rpc_url=https%3A%2F%2Frpc-mainnet.matic.network`.
+
+`chain_name` is required to specify the name of the network to be added.
+
+`name` and `symbol` if provided, SHOULD be a human-readable string representing the native token.
+
+`decimals` if provided, MUST be a non-negative integer representing the decimal precision of the native token.
+
+`explorer_url` if provided, MUST specify one or more URLs pointing to block explorer web sites for the chain.
+
+`icon_url` if provided, MUST specify one or more URLs pointing to reasonably sized images that can be used to visually identify the chain.
+
+An example of adding a network with RPC endpoints `https://rpc-polygon.com` and `https://rpc-mainnet.matic.network`, the name `Polygon Mainnet`, token `Matic`, symbol `MATIC`, decimals `18`, explorer at `https://polygonscan.com/`, and Chain ID `137` would look as follows:
+
+```URL
+ethereum:network-add@137/?chain_name=Polygon%20Mainnet&rpc_url=https%3A%2F%2Frpc-polygon.com&rpc_url=https%3A%2F%2Frpc-mainnet.matic.network&name=Matic&symbol=MATIC&decimals=18&explorer_url=https%3A%2F%2Fpolygonscan.com
+```
+
+## Rationale
+
+In furtherance of the Ethereum URL saga, network configuration is a needed addition to the possibility of Ethereum URLs. This would improve functionality for URLs, and offer non-mainnet users a way to connect without needing to configure their wallet by hand.
+
+The URL follows [EIP-831](./eip-831.md) with the `PREFIX` being `network` and the `PAYLOAD` being a composite of `add` and [EIP-681](./eip-681.md)-like `chain_id` and parameters.
+
+The choice for `PREFIX` being `network` is to allow further expansion and allow variants following the pattern `network-x`.
+
+An example URL for adding the Optimism Network
+
+```URL
+ethereum:network-add@10/?chain_name=Optimistic%20Ethereum
+&rpc_url=https%3A%2F%2Fmainnet.optimism.io&name=Ethereum&symbol=ETH&decimals=18&explorer_url=https%3A%2F%2Foptimistic.etherscan.io
+```
+
+The specification allows for a multitude of `rpc_url` and `explorer_url` to be specified. This is done such to overlap with parsing of the `TYPE` mentioned in [EIP-681](./eip-681.md).
+
+## Security Considerations
+
+URLs can be malformed to deceive users. Users SHOULD confirm source of URL before using any links. As well as checking source and transaction details before confirming any transactions. Applications SHOULD display network config, prior to network addition, such that users can confirm the validity of the network configuration being added.
+
+## References
+
+**Standards**
+[EIP-831](./eip-831.md)
+[EIP-681](./eip-681.md)
+[EIP-2400](./eip-2400.md)
+
+**Keyword Practices**
+[RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119)
+
```

Comment:
```suggestion
```
This is not a valid section.  Please inline references to relevant standards, and don't link externally.